src/App.tsx

```diff
@@ -16,35 +26,121 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([...goodsFromServer]);
+  const [sortState, setSortState] = useState<{
+    reversed: boolean;
+    criterion: SortType;
+  }>({
+    reversed: false,
+    criterion: SortType.None,
+  });
+  const [isReset, setIsReset] = useState(false as boolean);
+
+  const sortGoodsByAlphabet: ClickButton = () => {
+    const sortedGoods: Good[] = [...goods].sort((a, b) => {
+      if (sortState.reversed) {
+        return b.localeCompare(a);
+      }
+
+      return a.localeCompare(b);
+    });
+
+    setGoods(sortedGoods);
+    setSortState({ ...sortState, criterion: SortType.Alphabet });
+    setIsReset(true);
+  };
+
+  const sortGoodsByLength: ClickButton = () => {
+    const sortedGoods = [...goods].sort((a, b) => {
+      if (sortState.reversed) {
+        return b.length - a.length;
+      }
+
+      return a.length - b.length;
+    });
+
+    setGoods(sortedGoods);
+    setSortState({ ...sortState, criterion: SortType.Length });
+    setIsReset(true);
+  };
+
+  const reverseGoods: ClickButton = () => {
+    setSortState(prevState => {
+      const reversed = !prevState.reversed;
+
+      setGoods(prevGoods => {
+        const newGoods = [...prevGoods].reverse();
+
+        setIsReset(
+          !newGoods.every(
+            (element, index) => element === goodsFromServer[index],
+          ),
+        );
+
+        return newGoods;
+      });
+
+      return { ...prevState, reversed };
+    });
+  };
+
+  const resetGoods: ClickButton = () => {
+    setGoods([...goodsFromServer]);
+    setSortState({
+      reversed: false,
+      criterion: SortType.None,
+    });
+    setIsReset(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={sortGoodsByAlphabet}
+          type="button"
+          className={`button is-info ${
+            sortState.criterion !== SortType.Alphabet && 'is-light'
```

Comment:
To set classes conditionally use [classnames](https://www.npmjs.com/package/classnames) or [clsx](https://www.npmjs.com/package/clsx) library